src/xml2_node.cpp

```diff
@@ -27,13 +27,57 @@ std::string nodeName(T* node, SEXP nsMap) {
   return prefix + ":" + name;
 }
 
+SEXP node_name_impl(SEXP x, SEXP nsMap) {
+  NodeType type = getNodeType(x);
+
+  SEXP out;
+
+  switch(type) {
+  case NodeType::missing:
+    out = NA_STRING;
+    break;
+  case NodeType::node: {
+    SEXP node_sxp = VECTOR_ELT(x, 0);
+    XPtrNode node(node_sxp);
+
+    std::string name = nodeName(node.checked_get(), nsMap);
+    out = Rf_mkCharLenCE(name.c_str(), name.size(), CE_UTF8);
+    break;
+  }
+  default: Rf_error("Unexpected node type");
+  }
+
+  return out;
+}
+
 // [[export]]
-extern "C" SEXP node_name(SEXP node_sxp, SEXP nsMap) {
+extern "C" SEXP node_name(SEXP x, SEXP nsMap) {
   BEGIN_CPP
-  XPtrNode node(node_sxp);
+  NodeType type = getNodeType(x);
+
+  switch(type)
+  {
+  case NodeType::missing:
+  case NodeType::node   :
```

Comment:
```suggestion
  case NodeType::node:
```

We don't typically try and align based on the `:` in switches